library/kafka_topics.py

```diff
@@ -0,0 +1,167 @@
+      default: 'digest'
+      choices: [digest, sasl]
+  zookeeper_auth_value:
+    description:
+      - 'when zookeeper is configured to use authentication, value used to '
+      - 'connect.'
+  zookeeper_ssl_check_hostname:
+    description:
+      - 'when using ssl for zookeeper, check if certificate for hostname is '
+      - 'correct.'
+    default: True
+  zookeeper_ssl_cafile:
+    description:
+      - 'when using ssl for zookeeper, content of ca cert file or path to '
+      - 'ca cert file.'
+  zookeeper_ssl_certfile:
+    description:
+      - 'when using ssl for zookeeper, content of cert file or path to '
+      - 'server cert file.'
+  zookeeper_ssl_keyfile:
+    description:
+      - 'when using ssl for zookeeper, content of keyfile or path to '
+      - 'server cert key file.'
+  zookeeper_ssl_password:
+    description:
+      - 'when using ssl for zookeeper, password for ssl_keyfile.'
+  zookeeper_sleep_time:
+    description:
+      - 'when updating number of partitions and while checking for'
+      - 'the ZK node, the time to sleep (in seconds) between'
+      - 'each checks.'
+      default: 5
+  zookeeper_max_retries:
+    description:
+      - 'when updating number of partitions and while checking for'
+      - 'the ZK node, maximum of try to do before failing'
+      default: 5
+  kafka_sleep_time:
+    description:
+      - 'when updating number of partitions and while checking for'
+      - 'kafka to applied, the time to sleep (in seconds) between'
+      - 'each checks.'
+      default: 5
+  kafka_max_retries:
+    description:
+      - 'when updating number of partitions and while checking for'
+      - 'kafka to applied, maximum of try to do before failing'
+      default: 5
+''' + DOCUMENTATION_COMMON
+
+EXAMPLES = '''
+
+    # creates a topic 'test' with provided configuation for plaintext
+    - name: create topics
+      kafka_topics:
+        topics:
+          - name: 'test'
+            partitions: 2
+            replica_factor: 1
+            options:
+              retention.ms: 574930
+              flush.ms: 12345
+            state: 'present'
+        zookeeper: >
+          "{{ hostvars['zk']['ansible_eth0']['ipv4']['address'] }}:2181"
+        bootstrap_servers: >
+          "{{ hostvars['kafka1']['ansible_eth0']['ipv4']['address'] }}:9092,
+          {{ hostvars['kafka2']['ansible_eth0']['ipv4']['address'] }}:9092"
+
+    # deletes a topic
+    - name: delete topic
+      kafka_lib:
+        topics:
+        - name: 'test'
+          state: 'absent'
+        zookeeper: >
+          "{{ hostvars['zk']['ansible_eth0']['ipv4']['address'] }}:2181"
+        bootstrap_servers: >
+          "{{ hostvars['kafka1']['ansible_eth0']['ipv4']['address'] }}:9092,
+          {{ hostvars['kafka2']['ansible_eth0']['ipv4']['address'] }}:9092"
+
+'''
+
+
+def main():
+    """
+    Module usage
+    """
+    spec = dict(
+        mark_others_as_absent=dict(type='bool', default=False),
+        topics=dict(
```

Comment:
Do you think we need to make sure the user is not giving a list with duplicates topic?

Reply:
I add some check before any modification.